xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -60,6 +60,8 @@
 
     protected static final String EMPTY_RECYCLE_BIN = "emptybin";
 
+    private static final String TO_RECYCLEBIN_PARAM = "toRecyclebin";
```

Comment:
hmm this is becoming an API, i.e. it augments the Standard URL Scheme of XWiki and thus we need to pick the name carefully since it's hard to change afterwards. Was there a proposal made on the forum? I can think of various options: a param to say if we should skip the recycle bin, a param to say if it should go in the recycle bin. 

Seconde note: what is "emptybin"? The parameter name doesn't seem to match with "toRecycleBin". If it were matching it would be "emptyRecycleBin", no?

Reply:
> hmm this is becoming an API, i.e. it augments the Standard URL Scheme of XWiki and thus we need to pick the name carefully since it's hard to change afterwards. Was there a proposal made on the forum? I can think of various options: a param to say if we should skip the recycle bin, a param to say if it should go in the recycle bin.

I like the `toRecycleBin` parameter. The part about API remains and we need to be careful to get an agreement when adding new public APIs. In general it's good to make a proposal for all apis to add in a thread in the dev forum just to make sure all the devs are ok.

Reply:
Proposal created : https://forum.xwiki.org/t/page-removal-http-api-update-option-to-delete-pages-permanently/7406